fvm/evm/emulator/emulator.go

```diff
@@ -306,6 +311,13 @@ func (bl *BlockView) DryRunTransaction(
 		txResult.GasConsumed += txResult.GasRefund
 	}
 
+	// call tracer on tx end
+	if proc.evm.Config.Tracer != nil &&
+		proc.evm.Config.Tracer.OnTxEnd != nil &&
+		txResult != nil {
```

Comment:
`txResult != nil` this should always be true

Reply:
I saw that `proc.run` can return `nil` for `txResult` in some cases:
```go
if err != nil {
	return nil, err
}
```
Besides, this is the exact same condition for both `RunTransaction` & `BatchRunTransaction`, so I wanted to be on the safe side.